modules/pipelines/worker.py

```diff
@@ -230,86 +244,108 @@ def worker(
             'end_frame_image_path': end_frame_image_path,
             'combine_with_source': combine_with_source,
             'num_cleaned_frames': num_cleaned_frames,
-            'save_metadata_checked': save_metadata_checked # Ensure it's in job_params for internal use
+            'save_metadata_checked': save_metadata_checked  # Ensure it's in job_params for internal use
         }
-        
+
         # Validate parameters
         is_valid, error_message = pipeline.validate_parameters(job_params)
         if not is_valid:
             raise ValueError(f"Invalid parameters: {error_message}")
-        
+
         # Prepare parameters
         job_params = pipeline.prepare_parameters(job_params)
-        
+
         if not high_vram:
             # Unload everything *except* the potentially active transformer
             unload_complete_models(text_encoder, text_encoder_2, image_encoder, vae)
-            if studio_module.current_generator is not None and studio_module.current_generator.transformer is not None:
-                offload_model_from_device_for_memory_preservation(studio_module.current_generator.transformer, target_device=gpu, preserved_memory_gb=8)
-
+            if current_generator is not None and current_generator.transformer is not None:
+                offload_model_from_device_for_memory_preservation(
+                    current_generator.transformer, target_device=gpu, preserved_memory_gb=settings.get("gpu_memory_preservation", 8.0))
```

Comment:
In the inherited FP demo code from Illyasviel, this was explicitly preserved at 8GB, unlike most preservations which used his Setting, which defaulted to 6GB. I don't know if this change is an issue, but it should be tested. (There's a second similar change below.)